cns/restserver/restserver.go

```diff
@@ -171,10 +173,79 @@ func (service *HTTPRestService) Start(config *common.ServiceConfig) error {
 	listener.AddHandler(cns.V2Prefix+cns.AttachContainerToNetwork, service.attachNetworkContainerToNetwork)
 	listener.AddHandler(cns.V2Prefix+cns.DetachContainerFromNetwork, service.detachNetworkContainerFromNetwork)
 
+	if err := service.CreateExtHnsNetwork(); err != nil {
+		log.Printf("[Azure CNS] Failed to create ext network. Error: %v", err)
+		return err
+	}
+
 	log.Printf("[Azure CNS]  Listening.")
 	return nil
 }
 
+// CreateExtHnsNetwork creates ext HNS network if not present.
+func (service *HTTPRestService) CreateExtHnsNetwork() error {
+	// Ext Hns network is valid for windows only
```

Comment:
the check should be outside the function, not inside.

Reply:
You are looking at the old update. This has been moved to the platform specific files.